tests/bluetooth/host_long_adv_recv/src/main.c

```diff
@@ -0,0 +1,476 @@
+
+	generate_sequence(&report_a_combined.data[0], report_a_combined.length, 'A', 'Z');
+	generate_sequence(&report_b_combined.data[0], report_b_combined.length, 'a', 'z');
+	generate_sequence(&report_c.data[0], report_c.length, '0', '9');
+
+	(void)memcpy(&report_a_1.data[0], &report_a_combined.data[0], report_a_1.length);
+	(void)memcpy(&report_a_2.data[0], &report_a_combined.data[report_a_1.length],
+		     report_a_2.length);
+
+	for (int i = 0; i < report_a_1_repeated.length; i += report_a_1.length) {
+		memcpy(&report_a_1_repeated.data[i], &report_a_1.data[0],
+		       MIN(report_a_1.length, (report_a_1_repeated.length - i)));
+	}
+
+	(void)memcpy(&report_b_1.data[0], &report_b_combined.data[0], report_b_1.length);
+	(void)memcpy(&report_b_2.data[0], &report_b_combined.data[report_b_1.length],
+		     report_b_2.length);
+
+	// Check that non-interleaved fragmented adv reports work
+	ztest_returns_value(get_expected_data, &report_a_combined.data[0]);
+	ztest_returns_value(get_expected_length, report_a_combined.length); // Expect a
+	ztest_returns_value(get_expected_data, &report_b_combined.data[0]);
+	ztest_returns_value(get_expected_length, report_b_combined.length); // Then b
+	send_adv_report(&report_a_1);
+	send_adv_report(&report_a_2);
+	send_adv_report(&report_b_1);
+	send_adv_report(&report_b_2);
+
+	// Check that legacy adv reports interleaved with fragmented adv reports work
+	ztest_returns_value(get_expected_data, &report_c.data[0]);
+	ztest_returns_value(get_expected_length, report_c.length); // Expect c
+	ztest_returns_value(get_expected_data, &report_a_combined.data[0]);
+	ztest_returns_value(get_expected_length, report_a_combined.length); // Then a
+	send_adv_report(&report_a_1);
+	send_adv_report(&report_c); // Interleaved legacy adv report
+	send_adv_report(&report_a_2);
+
+	// Check that complete adv reports interleaved with fragmented adv reports work
+	ztest_returns_value(get_expected_data, &report_b_2.data[0]);
+	ztest_returns_value(get_expected_length, report_b_2.length); // Expect b
+	ztest_returns_value(get_expected_data, &report_a_combined.data[0]);
+	ztest_returns_value(get_expected_length, report_a_combined.length); // Then a
+	send_adv_report(&report_a_1);
+	send_adv_report(&report_b_2); // Interleaved short extended adv report
+	send_adv_report(&report_a_2);
+
+	// Check that fragmented adv reports from one peer are received,
+	// and that interleaved fragmented adv reports from other peers are discarded
+	ztest_returns_value(get_expected_data, &report_a_combined.data[0]);
+	ztest_returns_value(get_expected_length, report_a_combined.length); // Expect only a
+	ztest_returns_value(get_expected_data, &report_b_2.data[0]);
+	ztest_returns_value(get_expected_length, report_b_2.length); // Expect b, INCOMPLETE REPORT
+	send_adv_report(&report_a_1);
+	send_adv_report(&report_b_1); // Interleaved fragmented adv report, NOT SUPPORTED
+	send_adv_report(&report_a_2);
+	send_adv_report(&report_b_2);
```

Comment:
Should I keep this to document the current behavior or remove it as interleaving is not currently supported? If interleaving was supported, `report_b_combined` would be expected instead of `report_b_2`.